clients/go/cmd/zbctl/internal/commands/root.go

```diff
@@ -166,23 +174,73 @@ func shouldOverwriteEnvVar(cliParam, envVar string) bool {
 	return cliParameterSet || !exists
 }
 
-func parseAddress() (address string, port string) {
-	address = DefaultAddressHost
+func parseAddress() (host string, port string, err error) {
+
+	if shouldUseAddress() {
+		return getHostPortFromAddress()
+	}
+	host, port = getHostPort()
+	return host, port, nil
+}
+
+func parseHostAndPortFromAddress(parsedAddress string) (string, string, error) {
+	host, port, err := net.SplitHostPort(parsedAddress)
+	if err != nil {
+		if strings.Contains(err.Error(), "missing port in address") {
+			host = parsedAddress
+			port = DefaultAddressPort
+			err = nil
+		}
+	}
```

Comment:
These conditions can be put into the same if
```suggestion
	if err != nil && strings.Contains(err.Error(), "missing port in address") {
		host = parsedAddress
		port = DefaultAddressPort
		err = nil
```

Reply:
Done